DiscordChatExporter.Core.Services/ExportService.cs

```diff
@@ -50,8 +50,21 @@ public ExportService(SettingsService settingsService, DataService dataService)
             await foreach (var message in _dataService.GetMessagesAsync(token, channel.Id, after, before, progress))
             {
                 // Add encountered users to the list of mentionable users
-                mentionableUsers.Add(message.Author);
-                mentionableUsers.AddRange(message.MentionedUsers);
+                var encounteredUsers = new List<User>();
+                encounteredUsers.Add(message.Author);
+                encounteredUsers.AddRange(message.MentionedUsers);
+
+                mentionableUsers.AddRange(encounteredUsers);
+
+                foreach (User u in encounteredUsers)
+                {
+                    if(!guild.Members.ContainsKey(u.Id))
+                    {
+                        var member = await _dataService.GetGuildMemberAsync(token, guild.Id, u.Id);
+                        guild.Members[u.Id] = member;
+                    }
+                }
+
```

Comment:
I'm thinking, maybe it will be easier at this point to just get all guild members? We can get them in batches of 1000 per request via API. That may be more efficient than making 1 request per each participant, and would probably make the code simpler too because we can remove `encounteredUsers` etc.

What do you think?

Reply:
I tried that as well - if you do that with a user token, Discord gives you a 403 forbidden and makes you reverify your email. I didn't test it with a bot token.

Reply:
Ouch